server/controllers/user.js

```diff
@@ -21,3 +24,42 @@ exports.searchUsers = asyncHandler(async (req, res, next) => {
 
   res.status(200).json({ users: users });
 });
+
+exports.uploadImage = asyncHandler(async (req, res, next) => {
+  if (!req.file) {
+    throw new Error("No file provided!");
+  }
+
+  const user = await User.findById(mongoose.Types.ObjectId(req.user.id));
+
+  if (!user) {
+    res.status(404);
+    throw new Error("No users found!");
+  }
+
+  if (user.imagePublicId !== "") {
+    await cloudinary.uploader.destroy(user.imagePublicId);
+  }
+
+  const uploadedImage = await cloudinary.uploader.upload(req.file.path, {
+    folder: "kanban",
+    resourc_type: "auto",
+  });
+
+  const { secure_url, public_id } = uploadedImage;
+  user.imageUrl = secure_url;
+  user.imagePublicId = public_id;
```

Comment:
I think a better approach is just setting the file name to the users id. Then when we are looking for duplicates we can just find by the id and not have to store an additional property on the DB.

Reply:
please can you explain this further? I don't understand.

Reply:
do you mean to manage the profile image as a separate model and attach the user's id?

Reply:
When we upload the profile photo to Cloudinary if we set the users id as the filename then when we need to upload it again we can check if that id exists in the CDN and then replace that image so we don't need to store an additional property.